hassio/entrypoint.sh

```diff
@@ -1,17 +1,29 @@
 #!/bin/sh
 
-/bin/mkdir /config/insteon-mqtt/
+/bin/cp /opt/insteon-mqtt/config-example.yaml /config/config.yaml.default
 
-/bin/cp /opt/insteon-mqtt/config-example.yaml /config/insteon-mqtt/config.yaml.default
+# Migrate add-on data from the Home Assistant config folder,
+# to the add-on configuration folder.
+# We add a file to the old directory so that hopefully a user
+# doesn't get confused in the future.
+if [ ! -f /config/config.yaml ] && [ -f /homeassistant/insteon-mqtt/config.yaml]; then
+    shopt -s dotglob
+    mv /homeassistant/insteon-mqtt/* /config/ \
+        || echo "Failed to migrate InsteonMQTT configuration"; exit 1;
+    cp /opt/insteon-mqtt/hassio/CONFIG-MOVED.txt /homeassistant/insteon-mqtt/CONFIG-MOVED.txt
+    sed -i "s/<<MIGRATION_DATE>>/$(date)/g" /homeassistant/insteon-mqtt/CONFIG-MOVED.txt
+    echo "InsteonMQTT configuration successfully migrated from /config/insteon-mqtt."
+fi
 
-if [ ! -f /config/insteon-mqtt/config.yaml ]; then
+if [ ! -f /config/config.yaml ]; then
     echo "Welcome to InsteonMQTT!"
     echo "Creating your initial config.yaml file."
-    /bin/cp /config/insteon-mqtt/config.yaml.default /config/insteon-mqtt/config.yaml
-    sed -i "s/#storage: 'data'/storage: '\/config\/insteon-mqtt\/data'/" /config/insteon-mqtt/config.yaml
-    sed -i "s/#file: \/var\/log\/insteon_mqtt.log/file: \/config\/insteon-mqtt\/insteon_mqtt.log/" /config/insteon-mqtt/config.yaml
-    echo "Please edit the file /config/insteon-mqtt/config.yaml"
+    /bin/cp /config/config.yaml.default /config/config.yaml \
+      || echo "Unable to create initial InsteonMQTT config.yaml file"; exit 1;
+    sed -i "s/#storage: 'data'/storage: '\/config\/insteon-mqtt\/data'/" /config/config.yaml
+    sed -i "s/#file: \/var\/log\/insteon_mqtt.log/file: \/config\/insteon_mqtt.log/" /config/config.yaml
```

Comment:
With the contents of the old `/config/insteon-mqtt` directory moved to `/config`, wouldn't that mean for legacy users, the data directory would now be in `/config/data`?  If so, wouldn't it be easier to support user errors/questions if new users have their data directories created in `/config/data` as well?

Also, sed will let you use an alternate character for the delimiter if you don't want to have to escape all of those forward-slashes.  For example, something like the following might be easier to read/maintain:

```suggestion
    sed -i "s|#storage: 'data'|storage: '/config/data'|" /config/config.yaml
    sed -i "s|#file: /var/log/insteon_mqtt.log|file: /config/insteon_mqtt.log|" /config/config.yaml
```

Reply:
Good catch.

The data directory is also copies over, not sure why there would be a need to use the old directory?

Reply:
Keeping the contents of the old `data` directory could save people the effort of refreshing the DB for all of their devices, no?

Reply:
Oh, I see what you are saying.

1. For new users, the data directory will be /config/data
2. For migrated users
- We move the entire contents of insteon-mqtt to config (this includes moving data)
- I think this is best, it seems like HomeAssistant may not allow access to the entire homeassistant_config folder for much longer
- Plus if we move the data to the addon config, it will get backed up with the addon

But you are right, we need to modify the users config to point to the new location.  I think it is best to do that automatically.  I doubt many, if any, have moved this location.

Reply:
OK, I pushed another commit that edits the location of the storage files and the logfile for users as long as they are using the default location.